sycl/source/detail/queue_impl.hpp

```diff
@@ -388,6 +388,20 @@ class queue_impl {
   template <typename Param>
   typename Param::return_type get_backend_info() const;
 
+  /// Provides a hint to the backend to execute previously issued commands on
+  /// this queue. Overrides normal batching behaviour. Note that this is merely
+  /// a hint and not a guarantee.
+  void flush() {
+    if (MGraph.lock()) {
+      throw sycl::exception(make_error_code(errc::invalid),
+                            "flush cannot be called for a queue which is "
+                            "recording to a command graph.");
+    }
+    if (!MQueues.empty()) {
+      getPlugin()->call<PiApiKind::piQueueFlush>(MQueues[0]);
```

Comment:
Why do we only flush the first queue? What about other queues? Do we even have a practical case when SYCL queue will encapsulate several underlying queues?

Reply:
Thanks, missed this, the comment next to the declaration of MQueues seems to suggest that multiple queues are neede for FPGA devices.